adagrams/game.py

```diff
@@ -1,13 +1,138 @@
+    'B': 3, 'C': 3, 'M': 3, 'P': 3,
+    'F': 4, 'H': 4, 'V': 4, 'W': 4, 'Y': 4,
+    'K': 5,
+    'J': 8, 'X': 8,
+    'Q': 10, 'Z': 10
+}
 def draw_letters():
-    pass
+    """
+    build a hand of 10 letters for the user
+    Parameters:    None
+    Returns:
+    a list of strings with single letters
+    """     
+    #step1: create a copy of the LETTER_POOL and save all keys in a list
+    letter_count_dict = {}
+    letter_list = []
+    for letter, count in LETTER_POOL.items():
+        letter_count_dict[letter] = count
+        letter_list.append(letter)
+    #step2:randomly pick 10 letters from the letter_pool: for each pick, update count, remove letter from letter_list if count gets to 0;  
+    hand = []
+    while len(hand) < 10:
+        index = randint(0, len(letter_list) - 1)
+        letter = letter_list[index]
+        count = letter_count_dict[letter] 
+        hand.append(letter)
+        letter_count_dict[letter] = count - 1
+        # new_count = count  - 1
+        if count == 1:
+            letter_list.pop(index)
+
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """
+    Check if an input word from a user only used characters that are within the hand list
+    Parameters:
+    - word as a string
+    - letter_bank as a list of drawn letters in a hand
+    Return: boolean
+    """
+    # create letter_freq_dict for letters in letter_bank
+    letter_freq_dict = {}
+    for letter in letter_bank:
+        if letter in letter_freq_dict:
+            letter_freq_dict[letter] += 1
+        else:
+            letter_freq_dict[letter] = 1
+    # iterate through each character in word string and check if char exists and compare counts; update counts when finding a matching
+    for char in word.upper():
+        if not char in letter_freq_dict or letter_freq_dict[char] == 0:
+            return False
+        else:
+            letter_freq_dict[char] -= 1
+    return True
 
 def score_word(word):
-    pass
+    """
+    Score the word from user input
+    Parameters: word as a string of chars
+    output: int (number of points)    
+    """
+    # create LETTER_VALUES as a global variable outside (see top)
+    # iterate through word string, find points for each char and add them up, check length for extra points
+    score = 0
+    for char in word.upper():
+        score += LETTER_VALUES[char]
+    if len(word) in [7, 8, 9, 10]:
+        score += 8
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    """
+    Find word with highest score
+    Parameter: a list of strings
+    Output: a tuple with a string and the score    
+    """
+    # calculate score for each word and store in word_scores dict, also update highest score
+    word_scores = {}
+    highest_score = 0
+    for word in word_list:
+        score = score_word(word)
+        word_scores[word] = score
+        if score > highest_score:
+            highest_score = score
```

Comment:
I like the dictionary you created that maps each word with its score! Also, I love the way you have separated out each step of this process. There are several different tie breakers that need to be taken into account and you've done a good job of tackling each of them one at a time instead of all at once!